src/azure/azureResourceController.ts

```diff
@@ -0,0 +1,96 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { SubscriptionClient, Location } from '@azure/arm-subscriptions';
+import { ResourceManagementClient, ResourceGroup } from '@azure/arm-resources';
+import { SqlManagementClient, Server } from '@azure/arm-sql';
+import * as mssql from 'vscode-mssql';
+import * as azureUtils from './utils';
+import { TokenCredentialWrapper } from "./TokenCredentialWrapper";
+
+export class AzureResourceController {
+	private _subscriptionClient: SubscriptionClient | undefined;
+	private _resourceManagementClient: ResourceManagementClient | undefined;
+	private _sqlManagementClient: SqlManagementClient | undefined;
+
+	public set SubscriptionClient(v: SubscriptionClient) {
```

Comment:
Same here - we shouldn't ever be setting the client on the entire controller like this.